### PR TITLE
Livechat extended REST APIs

### LINKED ISSUE #3957: Request for additional REST API endpoint for LiveChat
For WordPress LiveChat plugin we need multiple endpoints to enable basic functionality.

- [ ] API access point - OAuth or similar point where user can login and authorize plugin - exchange necessary API keys
- [ ] LiveChat options endpoint - enable changing of LiveChat options - mainly enable/disable LiveChat, appearance optional, since we can do that through LiveChat API.
- [ ] Users - assign users as menagers and agents, remove user from these groups
- [ ] Departments - create, edit, delete, add agents
- [ ] Triggers - create trigger - low priority

With these basic endpoints we can enable WP user to easily setup LiveChat. Goal is to minimize the need for user to leave WP when setting up the LiveChat. In the future user could setup/buy a complete RC instance through WP interface.

### PR DESCRIPTION
**Pull request by sampaiodiego**
@RocketChat/core 

Closes #3957

New endpoints examples:

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	http://localhost:3000/api/v1/settings/Livechat_enabled
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	-H "Content-Type: application/json" \
	-X POST \
	-d '{ "value": true }' \
	http://localhost:3000/api/v1/settings/Livechat_enabled
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	http://localhost:3000/api/v1/livechat/users/agent
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	-X POST -H "Content-type:application/json" \
	-d '{"username":"diego.sampaio"}' \
	http://localhost:3000/api/v1/livechat/users/agent
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	http://localhost:3000/api/v1/livechat/users/agent/EgbhyZi2cQe6AJXBG
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	-X DELETE \
	http://localhost:3000/api/v1/livechat/users/agent/EgbhyZi2cQe6AJXBG
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	http://localhost:3000/api/v1/livechat/users/manager
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	-X POST -H "Content-type:application/json" \
	-d '{"username":"diego.sampaio"}' \
	http://localhost:3000/api/v1/livechat/users/manager
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	http://localhost:3000/api/v1/livechat/users/manager/EgbhyZi2cQe6AJXBG
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	-X DELETE \
	http://localhost:3000/api/v1/livechat/users/manager/EgbhyZi2cQe6AJXBG
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	http://localhost:3000/api/v1/livechat/department
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	http://localhost:3000/api/v1/livechat/department/zNh6q7Rw9JXFdGhMq
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	-X POST -H "Content-type:application/json" \
	-d '{"department": {"enabled": false, "name": "new from api" }, "agents": [{ "agentId": "yhHvK7uhhXh9DqKWH", "username": "diego.sampaio" }] }' \
	http://localhost:3000/api/v1/livechat/department
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	-X PUT -H "Content-type:application/json" \
	-d '{"department": {"enabled": false, "name": "update api 2"}, "agents": [{ "agentId": "yhHvK7uhhXh9DqKWH", "username": "diego.sampaio", "count": 2 }] }' \
	http://localhost:3000/api/v1/livechat/department/Cb6DH9r9b9ssYtAG9
```

```
curl -H "X-Auth-Token: LOGIN_TOKEN" -H "X-User-Id: USER_ID" \
	-X DELETE \
	http://localhost:3000/api/v1/livechat/department/Cb6DH9r9b9ssYtAG9
```

### CODE REVIEW

**Comment by sampaiodiego** [comment]
I'm not sure if this is the best way to create this APIs though :disappointed:

**Comment by sampaiodiego** [comment]
Ready for merge!

Should we start create some tests for REST APIs?

File: packages/rocketchat-api/server/settings.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// settings endpoints
+RocketChat.API.v1.addRoute('settings/:_id', { authRequired: true }, {
+	get() {
+		if (!RocketChat.authz.hasPermission(this.userId, 'view-privileged-setting')) {
+			return RocketChat.API.v1.unauthorized();
+		}
+
+		return RocketChat.API.v1.success(_.pick(RocketChat.models.Settings.findOneNotHiddenById(this.urlParams._id), '_id', 'value'));
```
Comment (rodrigok) [inline packages/rocketchat-api/server/settings.js]: Why not use `RocketChat.settings.get`?

File: packages/rocketchat-livechat/package.js [hunk 2]
```diff
@@ -190,8 +191,10 @@ Package.onUse(function(api) {
 	api.addFiles('server/publications/livechatInquiries.js', 'server');
 	api.addFiles('server/publications/livechatOfficeHours.js', 'server');
 
-	// api
-	api.addFiles('server/api.js', 'server');
+	// REST endpoints
+	api.addFiles('server/rest/departments.js', 'server');
+	api.addFiles('server/rest/sms.js', 'server');
+	api.addFiles('server/rest/users.js', 'server');
```
Comment (rodrigok) [inline packages/rocketchat-livechat/package.js]: Maybe we can start using imports here. What do you think?
We can keep the `server/api.js` and just add the imports for each api file.

File: packages/rocketchat-api/server/settings.js (hunk 1 above)
Comment (sampaiodiego) [inline packages/rocketchat-api/server/settings.js]: Because I don't want let users see settings hidden from the admin panel.

**Pull request merged**